cms/djangoapps/contentstore/rest_api/v1/views/tests/test_xblock.py

```diff
@@ -0,0 +1,291 @@
+from unittest.mock import patch
+from django.http import JsonResponse
+
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+
+from cms.djangoapps.contentstore.tests.test_utils import AuthorizeStaffTestCase
+
+
+TEST_LOCATOR = "block-v1:dede+aba+weagi+type@problem+block@ba6327f840da49289fb27a9243913478"
+
+
+class XblockViewTestCase(AuthorizeStaffTestCase):
+    # assumes that you want to pass a block id to the url
+
+    def get_test_data(self):
+        raise NotImplementedError("get_test_data must be implemented by subclasses")
+
+    def get_url_params(self):
+        return {"course_id": self.get_course_key_string(), "usage_key_string": TEST_LOCATOR}
+
+    def get_url(self):
+        return reverse(
+            "cms.djangoapps.contentstore:v1:studio_content",
+            kwargs=self.get_url_params(),
+        )
+
+    def send_request(self):
+        raise NotImplementedError("send_request must be implemented by subclasses")
+
+    @patch(
+        "cms.djangoapps.contentstore.rest_api.v1.views.xblock.handle_xblock",
+        return_value=JsonResponse(
+            {
+                "locator": TEST_LOCATOR,
+                "courseKey": AuthorizeStaffTestCase.get_course_key_string(),
+            }
+        ),
+    )
+    @patch(
+        "cms.djangoapps.contentstore.rest_api.v1.views.xblock.toggles.use_studio_content_api",
+        return_value=True,
+    )
+    def make_request(
+        self,
+        mock_use_studio_content_api,
+        mock_handle_xblock,
+        run_assertions=None,
+        course_id=None,
+        data=None,
+    ):
+        url = self.get_url()
+        data = self.get_test_data()
+
+        response = self.send_request(url, data)
+
+        # run optional callback method with additional assertions
+        if run_assertions:
+            run_assertions(
+                response=response, mock_handle_xblock=mock_handle_xblock
+            )
+
+        return response
+
+
+class XblockViewGetTest(XblockViewTestCase, ModuleStoreTestCase, APITestCase):
+    """
+    Test GET operation on xblocks
+    """
+
+    def get_test_data(self):
+        return None
+
+    def assert_xblock_handler_called(self, *, mock_handle_xblock, response):
+        mock_handle_xblock.assert_called_once()
+        passed_args = mock_handle_xblock.call_args[0][0]
+
+        assert passed_args.method == "GET"
+        assert passed_args.path == self.get_url()
+
+    def send_request(self, url, data):
+        return self.client.get(url)
+
+    def test_api_behind_feature_flag(self):
```

Comment:
I think in this case the duplication is actually more transparent and obvious. I also don't want to change/refactor any more logic, because that always causes some new linter errors to pop up that are in turn hard to fix. Can we either just leave this as is, or defer it to a followup PR? I just prefer to merge as easily as possible. - I actually based the swagger-docs-branch on this branch, so I could refactor it in that (following) PR with much much less potential trouble.